src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -197,24 +197,12 @@ private void RevertAutoLogonSpecificSettings(string domainName, string userName)
                                                                             RegistryConstants.MachineSettings.SubKeys.AutoLogon, 
                                                                             RegistryConstants.MachineSettings.ValueNames.AutoLogonUserName);
 
-            if(string.Equals(actualDomainNameForAutoLogon, domainName, StringComparison.CurrentCultureIgnoreCase)
-                ||string.Equals(actualUserNameForAutoLogon, userName, StringComparison.CurrentCultureIgnoreCase))
+            if (string.Equals(actualDomainNameForAutoLogon, domainName, StringComparison.CurrentCultureIgnoreCase)
+                && string.Equals(actualUserNameForAutoLogon, userName, StringComparison.CurrentCultureIgnoreCase))
             {
-                RevertOriginalValue(RegistryHive.LocalMachine, 
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon, 
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogon);
-                RevertOriginalValue(RegistryHive.LocalMachine,
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonUserName);
-                RevertOriginalValue(RegistryHive.LocalMachine,
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonDomainName);
-                RevertOriginalValue(RegistryHive.LocalMachine,
```

Comment:
earlier the model was to take backup of existing settings during configuration of the agent.. I think it is over-kill and is causing issues in many scenarios (some of them I mentioned in the mail). now as part of unconfig I am just resetting the changes that were done during configuration.. AutoLogonPassword key doesn't play any role when AutoLogon is not enabled..